snuba/request/schema.py

```diff
@@ -125,3 +131,26 @@ def __generate_template_impl(self, schema) -> Any:
 
     def generate_template(self) -> Any:
         return self.__generate_template_impl(self.__composite_schema)
+
+
+SETTINGS_SCHEMA = {
```

Comment:
```suggestion
SETTINGS_SCHEMAS = {
```

Also, this gets inferred as `builtins.dict[abc.ABCMeta*, builtins.dict*[builtins.str, builtins.object]]`. (You can check with `reveal_type` like this:

```diff
diff --git a/snuba/request/schema.py b/snuba/request/schema.py
index d1a3ee6..62b2ed0 100644
--- a/snuba/request/schema.py
+++ b/snuba/request/schema.py
@@ -153,4 +153,6 @@ SETTINGS_SCHEMA = {
         "properties": {},
         "additionalProperties": False,
     },
-}
\ No newline at end of file
+}
+
+reveal_type(SETTINGS_SCHEMA)
```

This should probably be explicitly defined as `Mapping[Type[RequestSettings], Schema]`.